packages/core/vue/src/useLocalStorage.ts

```diff
@@ -0,0 +1,19 @@
+import { customRef, Ref } from '@vue/reactivity';
+
+export function useLocalStorage<T> (key: string, defaultValue: T | null = null): Ref<T | null> {
+    return customRef<T | null>((track, trigger) => ({
+        get: () => {
+            track();
+            const value = localStorage.getItem(key);
+            return value ? JSON.parse(value) : defaultValue;
+        },
+        set: value => {
+            if (value === null) {
+                localStorage.removeItem(key);
+            } else {
+                localStorage.setItem(key, JSON.stringify(value));
```

Comment:
Should probably do the same here, since `JSON.stringify` will throw for circular references.

Reply:
All done. Am I right to assume we want to fail silently if that happens?

Btw, there is already an implementation of `useLocalStorage` in the `@vueuse/core` npm package but I assumed we didn't want an extra dependency just for that. Let me know, if you'd rather I use it instead.

https://vueuse.org/core/useLocalStorage/

Reply:
I think it's okay to just fail to add the item. We might want to `console.error` though, wdyt?

Reply:
Good shout! 👍